msgpack module: centos7 build error

```
[   67s] Scanning dependencies of target msgpack
[   67s] [ 25%] Building C object CMakeFiles/msgpack.dir/src/mpack.c.o
[   67s] /home/abuild/rpmbuild/BUILD/qore-msgpack-module-9.99/src/mpack.c: In function 'mpack_str_check_no_null':
[   67s] /home/abuild/rpmbuild/BUILD/qore-msgpack-module-9.99/src/mpack.c:800:5: error: 'for' loop initial declarations are only allowed in C99 mode
[   67s]      for (size_t i = 0; i < bytes; ++i)
[   67s]      ^
[   67s] /home/abuild/rpmbuild/BUILD/qore-msgpack-module-9.99/src/mpack.c:800:5: note: use option -std=c99 or -std=gnu99 to compile your code
[   67s] /home/abuild/rpmbuild/BUILD/qore-msgpack-module-9.99/src/mpack.c: In function 'mpack_expect_enum':
[   67s] /home/abuild/rpmbuild/BUILD/qore-msgpack-module-9.99/src/mpack.c:3789:5: error: 'for' loop initial declarations are only allowed in C99 mode
[   67s]      for (size_t i = 0; i < count; ++i) {
[   67s]      ^
[   67s] /home/abuild/rpmbuild/BUILD/qore-msgpack-module-9.99/src/mpack.c: In function 'mpack_expect_enum_optional':
[   67s] /home/abuild/rpmbuild/BUILD/qore-msgpack-module-9.99/src/mpack.c:3822:5: error: 'for' loop initial declarations are only allowed in C99 mode
[   67s]      for (size_t i = 0; i < count; ++i) {
[   67s]      ^
[   67s] /home/abuild/rpmbuild/BUILD/qore-msgpack-module-9.99/src/mpack.c: In function 'mpack_node_map_int_impl':
[   67s] /home/abuild/rpmbuild/BUILD/qore-msgpack-module-9.99/src/mpack.c:5444:5: error: 'for' loop initial declarations are only allowed in C99 mode
[   67s]      for (size_t i = 0; i < node.data->len; ++i) {
[   67s]      ^
[   67s] /home/abuild/rpmbuild/BUILD/qore-msgpack-module-9.99/src/mpack.c: In function 'mpack_node_map_uint_impl':
[   67s] /home/abuild/rpmbuild/BUILD/qore-msgpack-module-9.99/src/mpack.c:5475:5: error: 'for' loop initial declarations are only allowed in C99 mode
[   67s]      for (size_t i = 0; i < node.data->len; ++i) {
[   67s]      ^
[   67s] /home/abuild/rpmbuild/BUILD/qore-msgpack-module-9.99/src/mpack.c: In function 'mpack_node_map_str_impl':
[   67s] /home/abuild/rpmbuild/BUILD/qore-msgpack-module-9.99/src/mpack.c:5509:5: error: 'for' loop initial declarations are only allowed in C99 mode
[   67s]      for (size_t i = 0; i < node.data->len; ++i) {
[   67s]      ^
[   67s] /home/abuild/rpmbuild/BUILD/qore-msgpack-module-9.99/src/mpack.c: In function 'mpack_node_enum_optional':
[   67s] /home/abuild/rpmbuild/BUILD/qore-msgpack-module-9.99/src/mpack.c:5608:5: error: 'for' loop initial declarations are only allowed in C99 mode
[   67s]      for (size_t i = 0; i < count; ++i) {
[   67s]      ^
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -35,6 +35,9 @@ else()
     message(FATAL_ERROR "The compiler ${CMAKE_CXX_COMPILER} has no C++11 support. Please use a different C++ compiler.")
 endif()
 
+# Set C99 mode for C (otherwise could not be compiled for example in Centos 7).
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -std=c99")
+
 set(QPP_SRC
     src/ql_msgpack.qpp
     src/QC_MsgPack.qpp

